### ISSUE TITLE
Several graphic issues with X68000 Akumajou Dracula

### ISSUE THREAD

**Issue by HazukiXX** [mamedev/mame#12773]
### MAME version

0.268

### System information

Windows 11 64b English, i9 14900K, Radeon 7900 XTX, 96GB RAM, Video output module: Auto.

### Incorrect behaviour

First of all, I know the game is marked as having imperfect graphics, but I thought of reporting where exactly those imperfections are anyway.

So, I did extensive testing across multiple versions of the emulator. Some emulate the game better than others, and newer ones aren't necessarily better. I'll list how things look in 3 version ranges: 0.244-0.254, 0.255-0.258 and 0.259-Current.

**0.244~0.254**

As old as those versions are, surprisingly the game saw its best emulation in them. They emulate it _almost_ perfectly, save for 2 issues:

1) The water from stages 1 and 2 have the wrong color. It actually matches how they look in the Playstation port, which probably also got that color due to inaccurate emulation. They're supposed to be a pretty dark green, as shown in those real hardware captures: 
https://www.youtube.com/watch?v=tFRuANqTiMw
https://www.youtube.com/watch?v=5bmGKl64Y2U

XM6 Type G (the most accurate X68000 emulator) emulates it correctly. Below are some comparison screenshots (left is MAME and right is XM6):

![1](https://github.com/user-attachments/assets/31b56585-d286-487e-917c-9635393f106f)
![2](https://github.com/user-attachments/assets/b10c0f0a-3739-414c-9084-8600156a2afb)
![3](https://github.com/user-attachments/assets/68b1e5b0-4783-47b6-bc19-c88f1334e0a9)
![4](https://github.com/user-attachments/assets/6d7d0c8c-8523-4316-b714-c50850a663a3)

2) The bottom row of pixels is glitchy. Most of the time it displays a black line when it's supposed to have graphics as well. Sometimes the emulator tries to load those graphics and they appear for only 1 frame, before reverting to the black line. This is especially noticeable in the werewolf battle room, where you can see Simon climbing the stairs through that line, which looks pretty silly.

![1](https://github.com/user-attachments/assets/8d4bd46c-0f88-4a88-946d-59cad7ec3717)

**0.255~0.258**

Version 0.255 introduced another bug on top of those two. Some columns of pixels were incorrectly shifted to the left. This is easily noticeable at the beginning of the first stage when Simon enters the castle. You can see part of his body through the rightmost column. Also, after killing a boss, its icon gets replaced with a cross in the castle map. That cross is supposed to be centered in the boss' square, but here they appear shifted to the left as well.

![1](https://github.com/user-attachments/assets/7cebb105-e0a4-4394-89f0-f29ea80f0bd2)
![2](https://github.com/user-attachments/assets/fcc609aa-00c6-479a-951e-4e5247bb5bda)

I imagine this bug was caused by the "-x68k: fix bg layer positioning and another layer selection case [cracyc]" change.

**0.259~Current**

0.259 fixed the above bug, but introduced a different one. Now some rows of pixels were shifted upwards. You can see it at the opening cutscene (pay attention to the bottom of the big candle) and in other places like the floor at the beginning of stage 1, or the crosses from the boss icons. I believe this was also due to one of those changes made by cracyc.

![1](https://github.com/user-attachments/assets/e76ae87a-17bc-418b-93e2-fda73af7f063)
![2](https://github.com/user-attachments/assets/636eebf1-a430-46af-bc96-8f2e8cb9800b)
![3](https://github.com/user-attachments/assets/fc6fbdaa-713c-4883-9f93-5f580c594511)

To sum it up, if you guys could revert to how the game looked in 0.244~0.254, plus fix the bottom row of pixels and the color of water, everything would be perfect.

### Expected behaviour

See above.

**Comment by cuavas** [comment]
@cracyc this seems to be related to a change you made.  Can you have a look?

**Comment by cracyc** [comment]
https://github.com/mamedev/mame/commit/6a4b1b300cb02e7d173f78fe7016473caf350dac
Dividing the color by 4 rather than 2 then adding makes it look more like the hardware pics.

**Comment by HazukiXX** [comment]
@cracyc thanks for the fixes, however there's still one issue left to address: the bottom row of pixels. Here are a couple zoomed-in
 comparison pics showing how it currently is (left) and how it's supposed to be (right):

![1](https://github.com/user-attachments/assets/e3d4b581-3c52-4e82-a9f7-ffc62273da21)
![2](https://github.com/user-attachments/assets/6c57f7ed-989a-40b8-b391-c52090614e52)

At stage 1's sewers for example, there's no water at the bottom row when there should be. And at the werewolf boss room, Simon's body can be seen through the floor (also note the additional yellow tiles detail missing from it).

**Comment by b0x-Cub3d** [comment]
It looks like you need to compress the sprite layer by 1 line. That's just my guess, though.

**Comment by cracyc** [comment]
https://github.com/mamedev/mame/commit/bfc648f2b1c2e0d106659968a3042c73ec8e7991 fixes this issue with strider but doesn't for this, probably a related problem though where the vblank irq is firing too early.

**Comment by b0x-Cub3d** [comment]
My bad, when I said "compress" I meant by "translating it down by 1 pixel".

**Comment by HazukiXX** [comment]
> [bfc648f](https://github.com/mamedev/mame/commit/bfc648f2b1c2e0d106659968a3042c73ec8e7991) fixes this issue with strider but doesn't for this, probably a related problem though where the vblank irq is firing too early.

This update made the bottom line from Dracula worse, causing visual artifacts to show up:
![1](https://github.com/user-attachments/assets/92ec3911-b6ac-470a-a532-f99cc49ba436)
![2](https://github.com/user-attachments/assets/c6b270de-bdbb-4748-9313-d0a005d618ae)

(You may need to open the couple screenshots in a new tab to see the glitches better)

**Comment by b0x-Cub3d** [comment]
I think those artifacts aren't meant to be seen on an actual CRT. So, it's actually more accurate in my opinion.

**Comment by cracyc** [comment]
https://github.com/mamedev/mame/commit/efa2fb25be9a824767b5bd506496a903ea864793 should fix this

**Comment by b0x-Cub3d** [comment]
Now, I have an issue with getting screenshots of X68000 games. It looks like they are vertically stretched.

**Comment by FoxhackDN** [comment]
You sure the screenshot aspect ratio is set correctly?

**Comment by b0x-Cub3d** [comment]
Yes, I tested 4:3 and 3:2. I still get the screenshot being vertically stretched.

**Comment by HazukiXX** [comment]
> [efa2fb2](https://github.com/mamedev/mame/commit/efa2fb25be9a824767b5bd506496a903ea864793) should fix this

There's no difference at all. Bottom line is still the same as before at least for me.
